src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -3223,14 +3225,32 @@ private void CreateContainerProcessInternal()
                     // Create PowerShell process inside the container.
                     //
                     result = HcsCreateProcess(ComputeSystem, cmd, ref ProcessInformation, ref Process, ref resultString);
-                    if (result != 0)
+                    if (result == 0)
                     {
-                        processId = 0;
-                        error = result;
+                        processId = Convert.ToInt32(ProcessInformation.ProcessId);
+                    }
+                    else if (result == 2147942402) // 0x80070002 (2147942402) - The system cannot find the file specified.
+                    {
+                        // If we cannot find PowerShell Core, fall back to `powershell.exe` or Windows Powershell
+                        // We cannot do "cmd.exe /c powershell" because it returns process id of cmd instead powershell/pwsh.
+                        Executable = "powershell.exe";
+                        cmd = GetContainerProcessCommand();
+
+                        result = HcsCreateProcess(ComputeSystem, cmd, ref ProcessInformation, ref Process, ref resultString);
```

Comment:
Seems like this can be simplified as a loop (having only one `HcsCreateProcess` and checking of the error).  Have `Executable` as an array containing `pwsh.exe`, `powershell.exe`.  Once process creation succeeds, break out of the loop, otherwise try the next executable.